src/components/product-gallery/ProductGalleryUI.tsx

```diff
@@ -0,0 +1,66 @@
+import { Footer } from "../base-gallery/Footer";
+import { FooterSubmitButton } from "../base-gallery/FooterSubmitButton";
+import { Form } from "../base-gallery/Form";
+import { FormattedMessage } from "react-intl";
+import { Header } from "../base-gallery/Header";
+import { HeaderSearchInput } from "../base-gallery/HeaderSearchInput";
+import {
+  HeaderSortProductsDropdown,
+  SortingProductsPair,
+} from "./HeaderSortProductsDropdown";
+import { Content } from "../base-gallery/Content";
+import { ProductGalleryContentEmpty } from "./ProductGalleryContentEmpty";
+import { ProductGalleryContentNoResult } from "./ProductGalleryContentNoResult";
+
+// TODO: implement it
+export const ProductGalleryUI = <TValue,>({
```

Comment:
`ProductGalleryUI` no debería ser genérico.

```suggestion
export const ProductGalleryUI = ({
```